### PR TITLE
Avoid unnecessary allocations when freeing stable structs

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:55ec12c0

---

**Stack**:
- #1734
- #1722
- #1704
- #1700 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: plugins/cairo-lang-macro/src/types/conversion.rs [hunk 1]
```diff
@@ -109,11 +101,8 @@ impl TextSpan {
...
 
     #[doc(hidden)]
-    pub fn from_owned_stable(span: StableTextSpan) -> Self {
-        Self {
-            start: span.start,
-            end: span.end,
-        }
+    pub fn free_owned_stable(span: StableTextSpan) {
+        let _ = span;
     }
```
Comment (wawel37) [inline plugins/cairo-lang-macro/src/types/conversion.rs:106]: Shouldn't this be in the StableTextSpan's impl?

File: scarb/src/compiler/plugin/proc_macro/ffi.rs [hunk 2]
```diff
@@ -147,8 +147,8 @@ impl ProcMacroInstance {
         // Free the memory allocated by the `stable_token_stream`.
         // This will call `CString::from_raw` under the hood, to take ownership.
         unsafe {
-            TokenStream::from_owned_stable(stable_result.input);
-            TokenStream::from_owned_stable(stable_result.input_attr);
+            TokenStream::free_owned_stable(stable_result.input);
```
Comment (wawel37) [inline scarb/src/compiler/plugin/proc_macro/ffi.rs:150]: Change comment in [this](https://github.com/software-mansion/scarb/blob/308f593261681a0c03a718ae3f985257fab34c67/scarb/src/compiler/plugin/proc_macro/ffi.rs#L136C53-L136C70) line, from `from_owned_stable` to `free_owned_stable`

**Review by wawel37 — APPROVED** [review]
Looks good. I was thinking about calling the [std::mem::drop](https://doc.rust-lang.org/std/mem/fn.drop.html) explicitly, but I don't think that's necessary

File: plugins/cairo-lang-macro/src/types/conversion.rs (hunk 1 above)
Comment (maciektr) [reply plugins/cairo-lang-macro/src/types/conversion.rs:106]: We implement all conversions in `cairo-lang-macro` not `cairo-lang-macro-stable`,  so the latter only defines the binary representation that is passed through the ffi barrier (so it changes as rarely as possible).

**Review by wawel37 — APPROVED** [review]

**Pull request merged**